src/transxchange/read.rs

```diff
@@ -549,7 +570,14 @@ fn calculate_stop_times(
             .ok_or_else(|| format_err!("stop_id={:?} not found", stop_point_ref))?;
         let stop_point_wait_from = get_duration_from(&stop_point, "WaitTime");
         let run_time = get_duration_from(&journey_pattern_timing_link, "RunTime");
-        let (pickup_type, drop_off_type) = get_pickup_and_dropoff_types(&stop_point, "Activity");
+        let stop_point_activity = get_stop_point_activity(
+            &journey_pattern_timing_link,
+            &vehicle_journey_timing_links,
+            &stop_point,
+            "From",
+        );
+        let (pickup_type, drop_off_type) =
+            get_pickup_and_dropoff_types(&stop_point_activity, "Activity");
```

Comment:
seen with @ArnaudOggy it will be called with `WaitTime`

Reply:
Indeed, I don't think so. Because `Activity` is very specific to the pickup and dropoff concepts. I guess this parameter could be removed and hardcoded in the function `get_pickup_and_dropoff_types()`.